src/coreclr/jit/lclvars.cpp

```diff
@@ -2587,14 +2587,16 @@ void Compiler::lvaSetVarLiveInOutOfHandler(unsigned varNum)
         {
             noway_assert(lvaTable[i].lvIsStructField);
             lvaTable[i].lvLiveInOutOfHndlr = 1;
-            if (!lvaEnregEHVars)
+            // For now, only enregister an EH Var if it is a single def.
+            if (!lvaEnregEHVars || !lvaTable[i].lvSingleDef)
```

Comment:
How trustworthy is `lvSingleDef`? 

I see it being set over in `lvaMarkLclRefs` and that should have run not too long before LSRA if I remember correctly. Would be good to verify this.

I suspect it may not be entirely accurate (it may have both false positive and false negatives). So long as we're only using it as a profitability screen here that's probably ok.

Reply:
Yes - Even Carol mentioned it and I tried looking around, but didn't find much clue that proves it might not be safe. I will do some more research and sync up offline with you if any questions.

Reply:
I believe that it is only set for compiler temps (lvTemp) and it should always be correct for those variables.

Reply:
Well that was the original implementation, but it has been extended to in fgFindJumpTargets to handle additional "root method locals"

```
    // Now that we've seen the IL, set lvSingleDef for root method
    // locals.
```